helm_lib/templates/_module_init_container.tpl

```diff
@@ -15,3 +15,19 @@
   - name: {{ $volume_name }}
     mountPath: /tmp/{{ $volume_name }}
 {{- end }}
+
+{{- /* Usage: {{ include "helm_lib_module_init_container_check_linux_kernel" (list . ">= 4.9.17") }} */ -}}
+{{- /* returns initContainer which checks the kernel version on the node for compliance to semver constraint */ -}}
+{{- define "helm_lib_module_init_container_check_linux_kernel"  }}
+  {{- $context := index . 0 -}}
+  {{- $semver_constraint := index . 1  -}}
+- name: check-linux-kernel
+  image: {{ include "helm_lib_module_common_image" (list $context "checkKernelVersion") }}
+  env:
+  - name: KERNEL_CONSTRAINT
+    value: {{ $semver_constraint | quote }}
+  securityContext:
+    runAsNonRoot: false
+    runAsUser: 0
```

Comment:
Is that really necessary?

Reply:
The root user, I mean.

Reply:
good point, init container should run from user described in security context for pod